### PR TITLE
enable windows build

### PR DESCRIPTION
**Pull request by kejxu**
also include removing exposed `windows.h` from header files

### CODE REVIEW

File: kobuki_node/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,9 @@ catkin_package(
                   ecl_exceptions ecl_sigslots ecl_streams ecl_threads
 )
 
+# debug
```
Comment (ooeygui) [inline kobuki_node/CMakeLists.txt]: Remove comment

File: kobuki_keyop/src/keyop_core.cpp [hunk 2]
```diff
@@ -288,6 +328,55 @@ void KeyOpCore::processKeyboardInput(char c)
    * the last one for its actual purpose (e.g. left arrow corresponds to
    * esc-[-D) we can keep the parsing simple.
    */
+#if defined(_WIN32)
```
Comment (ooeygui) [inline kobuki_keyop/src/keyop_core.cpp]: The Duplicated handling feels out of place. Let's see if we can only have one switch statement which moves the cases into a block which switches between Win32 and Linux.

Comment (kejxu) [reply kobuki_keyop/src/keyop_core.cpp]: added `unordered_map` for virtual key code remapping

File: kobuki_keyop/include/keyop_core/keyop_core.hpp [hunk 3]
```diff
@@ -109,15 +99,30 @@ class KeyOpCore
   /*********************
    ** Keylogging
    **********************/
-
   void keyboardInputLoop();
   void processKeyboardInput(char c);
   void remoteKeyInputReceived(const kobuki_msgs::KeyboardInput& key);
   void restoreTerminal();
+
```
Comment (ooeygui) [inline kobuki_keyop/include/keyop_core/keyop_core.hpp:119]: Avoid code cleanup in porting

File: kobuki_node/launch/minimal.launch [hunk 4]
```diff
@@ -5,6 +5,7 @@
   <arg name="kobuki_publish_tf" default="true"/> <!-- Publish base_footprint - odom transforms (usually good thing to have for localisation) -->
 
   <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
+
```
Comment (ooeygui) [inline kobuki_node/launch/minimal.launch]: Avoid Code Cleanup

File: kobuki_node/src/library/kobuki_ros.cpp [hunk 5]
```diff
@@ -194,8 +194,7 @@ bool KobukiRos::init(ros::NodeHandle& nh, ros::NodeHandle& nh_pub)
     }
     else
     {
-      ROS_INFO_STREAM("Kobuki : configured for connection on device_port "
-                      << parameters.device_port << " [" << name << "].");
+      ROS_INFO_STREAM("Kobuki : configured for connection on device_port " << parameters.device_port << " [" << name << "].");
```
Comment (ooeygui) [inline kobuki_node/src/library/kobuki_ros.cpp]: Avoid code cleanup

**Review by ooeygui — APPROVED** [review]

**Review by seanyen — APPROVED** [review]
LGTM

**Pull request merged**